### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by jschueller**
- #1564
- fix quadratic coeff in QuadraticBasisFactory to get x² term instead of x²/2
- A fix for the input dimension of the likelihood function from GLM/Kriging (bug introduced fixing #1455)
- Need to rerun GLM after bounds change
- Cleanup the Path class API
- Change VisualTest::DrawLinearModel colors: swap red/blue for the diagonal/clouds for consistency with DrawHenryLine etc

### CODE REVIEW

**Comment by jschueller** [comment]
@regislebrun I plan to add more fixes as they come, that's why this PR is marked "draft" (you can see the color: it's gray), when it's ready to review I'll mark it as "ready" (it goes green)

**Comment by josephmure** [comment]
@jschueller  I am currently rewriting the `SobolIndicesExperiment` doc page as part of #1562. Could you please delete commit 64514f6 so as to avoid merge conflicts?

**Comment by jschueller** [comment]
> @jschueller I am currently rewriting the `SobolIndicesExperiment` doc page as part of #1562. Could you please delete commit [64514f6](https://github.com/openturns/openturns/commit/64514f675bc555c0212551e58c2a04f02562405e) so as to avoid merge conflicts? The rewrite is going to fix #1567 anyway.

ok

**Comment by jschueller** [comment]
This is ready to go

File: python/doc/examples/calibration/calibration_withoutobservedinputs.ipynb [hunk 1]
```diff
@@ -0,0 +1,452 @@
+{
```
Comment (josephmure) [inline python/doc/examples/calibration/calibration_withoutobservedinputs.ipynb:1]: Could you please convert this example to a Python script with RST commands that could work with #1574? You can use [this file](https://github.com/openturns/openturns/blob/0224e7b581878431e52e5af357def2fbdbc5d9b4/python/doc/examples/py_calibration/bayesian_calibration/plot_bayesian_calibration.py) as template.

File: python/src/LinearBasisFactory_doc.i.in [hunk 2]
```diff
@@ -13,15 +16,11 @@ See also
...
--------
-basis : :class:`~openturns.Basis`."
+Examples
+--------
+>>> import openturns as ot
+>>> basis = LinearBasisFactory(2).build()
+>>> f = ot.AggregatedFunction(basis)
+>>> x = [2, 3]
+>>> print(f(x))
+[1,2,3]"
```
Comment (josephmure) [inline python/src/LinearBasisFactory_doc.i.in:26]: Wouldn't it be better to print the basis? Isn't there a pretty print that could be used?

File: python/src/LinearModelAlgorithm_doc.i.in [hunk 3]
```diff
@@ -42,11 +42,11 @@ If basis is not specified, the underlying model is :
 
     \tilde{Y} = a_0 + \sum_{i=1}^n a_i X_i + \epsilon
 
-The coefficients :math:`a_i` are evaluated using a least squares method. Default
- method is `QR`. User might choose also `SVD` or `Cholesky` (usefull if basis
- is orthogonal) and large dataset.
+The coefficients :math:`a_i` are evaluated using a least squares method.
+Default method is `QR`. User might choose also `SVD` or `Cholesky` (useful
+if basis is orthogonal) and large dataset.
```
Comment (josephmure) [inline python/src/LinearModelAlgorithm_doc.i.in:47]:
```suggestion
if the basis is orthogonal) and large dataset.
```

File: python/src/LinearModelAlgorithm_doc.i.in [hunk 4]
```diff
@@ -42,11 +42,11 @@ If basis is not specified, the underlying model is :
 
     \tilde{Y} = a_0 + \sum_{i=1}^n a_i X_i + \epsilon
 
-The coefficients :math:`a_i` are evaluated using a least squares method. Default
- method is `QR`. User might choose also `SVD` or `Cholesky` (usefull if basis
- is orthogonal) and large dataset.
+The coefficients :math:`a_i` are evaluated using a least squares method.
+Default method is `QR`. User might choose also `SVD` or `Cholesky` (useful
```
Comment (josephmure) [inline python/src/LinearModelAlgorithm_doc.i.in:46]:
```suggestion
Default method is `QR`. User might also choose `SVD` or `Cholesky` (useful
```
Seems more natural to me.

File: python/src/LinearModelAlgorithm_doc.i.in [hunk 5]
```diff
@@ -42,11 +42,11 @@ If basis is not specified, the underlying model is :
...
+if basis is orthogonal) and large dataset.
 
-The evaluation of the coefficients is completed by some usefull parameters that could
+The evaluation of the coefficients is completed by some useful parameters that could
 help the diagnostic of the linearity."
```
Comment (josephmure) [inline python/src/LinearModelAlgorithm_doc.i.in:50]: > The evaluation of the coefficients is completed by some useful parameters that could
help the diagnostic of the linearity.

It is good that you corrected the typo, but I don't understand what this sentence means.

File: python/src/QuadraticBasisFactory_doc.i.in [hunk 6]
```diff
@@ -13,15 +16,11 @@ See also
...
-"Build the basis.
-
-Returns
--------
-basis : :class:`~openturns.Basis`."
+Examples
+--------
+>>> import openturns as ot
+>>> basis = QuadraticBasisFactory(2).build()
+>>> f = ot.AggregatedFunction(basis)
```
Comment (josephmure) [inline python/src/QuadraticBasisFactory_doc.i.in:23]: Same as before: wouldn't it be better to simply print the basis?

File: python/src/ConstantBasisFactory_doc.i.in [hunk 7]
```diff
@@ -15,13 +12,13 @@ BasisFactory, LinearBasisFactory, QuadraticBasisFactory
...
-"Build the basis.
-
-Returns
--------
-basis : :class:`~openturns.Basis`."
+Examples
+--------
+>>> import openturns as ot
+>>> basis = ConstantBasisFactory(2).build()
+>>> f = ot.AggregatedFunction(basis)
```
Comment (josephmure) [inline python/src/ConstantBasisFactory_doc.i.in:21]: Wouldn't it be better to simply print the basis?

File: python/src/SobolIndicesAlgorithmImplementation_doc.i.in [hunk 8]
```diff
@@ -45,8 +45,8 @@ First and second order Sobol' indices are defined as follows:
 .. math::
 
    \begin{array}{ccc}
-    S_i & = & \frac{V_i}{\Var{\vect{Y}}} \\
-    S_{i,j} & = & \frac{V_{i,j}}{\Var{\vect{Y}}}
+    S_i & = & \frac{V_i}{\Var{Y}} \\
```
Comment (josephmure) [inline python/src/SobolIndicesAlgorithmImplementation_doc.i.in:48]: Great that you included this correction!

**Review by josephmure — CHANGES_REQUESTED** [review]
My main comment concerns the new calibration example with no inputs: could you please produce it as a Python file in order to fit #1574, using [this](https://github.com/openturns/openturns/blob/0224e7b581878431e52e5af357def2fbdbc5d9b4/python/doc/examples/py_calibration/bayesian_calibration/plot_bayesian_calibration.py) as a template?

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**